src/client/Microsoft.Identity.Client/ApiConfig/Executors/ManagedIdentityExecutor.cs

```diff
@@ -35,19 +38,48 @@ public ManagedIdentityExecutor(IServiceBundle serviceBundle, ManagedIdentityAppl
         {
             var requestContext = CreateRequestContextAndLogVersionInfo(commonParameters.CorrelationId, cancellationToken);
 
+            if (_managedIdentityApplication.KeyMaterialManager.CryptoKeyType != CryptoKeyType.None)
+            {
+                // managed identity resource 
+                string miResource = managedIdentityParameters.Resource;
+
+                // Check if the input ends with "/.default"
+                bool endsWithDefault = miResource.EndsWith("/.default", StringComparison.OrdinalIgnoreCase);
+
+                // Add "/.default" only if it doesn't end with "/.default" and doesn't contain "/.somethingelse"
```

Comment:
Is this part of the comment accurate? `and doesn't contain "/.somethingelse"`

Reply:
Indeed, I am suprpised to see this. 

In any case, as @neha-bhargava mentioned, this check is done elsewhere already. MSAL should work with both resource/.default and resource.

Reply:
For MSI endpoints we remove /.default suffix, when using ests the /.default suffix needs to be added if not already there. Not sure about the possibility of ./somethingelse